### PR TITLE
Overhaul of docker building

### PR DESCRIPTION
**Pull request by kormat**
- Simplify down to a single image
- Trimmed the image from ~600MB down to 466MB
- Move package lists to docker/ text files
- Pre-build crypto libs, and stop building python2 versions
- Fix issue with dir timestamp that caused docker to rebuild more than
  necessary
- Updated docker/README.md
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/208%23issuecomment-117570883%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33727069%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33727208%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23issuecomment-117832346%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33753933%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33753994%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23issuecomment-117951697%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23issuecomment-117570883%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40pszalach%20%40shitz%20%5Cr%5CnI%27m%20happy%20to%20talk%20%20you%20through%20any%20docker-specific%20stuff%2C%20a%20lot%20of%20this%20is%20more%20general%20though.%22%2C%20%22created_at%22%3A%20%222015-07-01T09%3A51%3A20Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Thanks%20for%20comments%21%20Looks%20good%20to%20me.%22%2C%20%22created_at%22%3A%20%222015-07-01T21%3A40%3A03Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40pszalach%20-%20cheers%20for%20the%20review%20%3A%29%22%2C%20%22created_at%22%3A%20%222015-07-02T08%3A05%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20762cb8480a65023671e5263c842eec00fe0b9db7%20docker/README.md%2040%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33727208%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%5C%22%29%5C%22%20is%20missing%22%2C%20%22created_at%22%3A%20%222015-07-01T21%3A33%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22Fixed%22%2C%20%22created_at%22%3A%20%222015-07-02T07%3A51%3A54Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20docker/README.md%3AL35-40%22%7D%2C%20%22Pull%20762cb8480a65023671e5263c842eec00fe0b9db7%20Dockerfile%2086%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/208%23discussion_r33727069%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22do%20we%20need%20screen%20at%20all%3F%22%2C%20%22created_at%22%3A%20%222015-07-01T21%3A32%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%2C%20%7B%22body%22%3A%20%22As%20you%20only%20get%20a%20single%20terminal%20into%20docker%2C%20not%20having%20screen%20severely%20limits%20your%20options%20when%20trying%20to%20debug%20stuff.%22%2C%20%22created_at%22%3A%20%222015-07-02T07%3A51%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20Dockerfile%3AL1-74%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/208#issuecomment-117570883'>General Comment</a></b>
- <a href='https://github.com/kormat'></a> @pszalach @shitz
  I'm happy to talk  you through any docker-specific stuff, a lot of this is more general though.
- <a href='https://github.com/pszalach'></a> Thanks for comments! Looks good to me.
- <a href='https://github.com/kormat'></a> @pszalach - cheers for the review :)
- [ ] <a href='#crh-comment-Pull 762cb8480a65023671e5263c842eec00fe0b9db7 Dockerfile 86'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/208#discussion_r33727069'>File: Dockerfile:L1-74</a></b>
- <a href='https://github.com/pszalach'></a> do we need screen at all?
- <a href='https://github.com/kormat'></a> As you only get a single terminal into docker, not having screen severely limits your options when trying to debug stuff.
- [ ] <a href='#crh-comment-Pull 762cb8480a65023671e5263c842eec00fe0b9db7 docker/README.md 40'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/208#discussion_r33727208'>File: docker/README.md:L35-40</a></b>
- <a href='https://github.com/pszalach'></a> ")" is missing
- <a href='https://github.com/kormat'></a> Fixed

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/208?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/208?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/208'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,57 +1,73 @@
...
 # Now copy over the current branch
-COPY scion.git /home/scion/scion.git
+COPY . $BASE/
+RUN sudo chown -R scion: $HOME
+# Build topology files
+RUN ./scion.sh topology
 # Copy over init.sh:
-COPY scion.git/docker/init.sh /home/scion/bin/
+COPY docker/init.sh $HOME/bin/
 # Install basic screen config
```
Comment (pszal) [inline Dockerfile]: do we need screen at all?

File: docker/README.md [hunk 2]
```diff
@@ -37,5 +35,5 @@ Notes:
  * As `docker.sh` copies your *working tree*, if you're trying to test a commit
    before sending for review/etc, make sure your working directory is clean
    before building the image.
- * Currently there is no way to view the generated sphinx documentation outside
-   of the image.
+ * Currently there is no way to view generated output (like code coverage, or
+   sphinx-docs outside of the image.
```
Comment (pszal) [inline docker/README.md]: ")" is missing

File: Dockerfile (hunk 1 above)
Comment (kormat) [inline Dockerfile]: As you only get a single terminal into docker, not having screen severely limits your options when trying to debug stuff.

File: docker/README.md (hunk 2 above)
Comment (kormat) [inline docker/README.md]: Fixed

**Pull request merged**